lib/network_utils.ml

```diff
@@ -913,13 +927,16 @@ end = struct
     let set_dns =
```

Comment:
* `set_dns` is not a good name because this returns a configuration and does not set anyting
* the code could be made flatter which I think is easier to read

```
let dns_config =
  let emit = List.mem (`dns interface) options in
  if emit && ipv6 then
    debug "%s is the DNS interface" interface ;
    ["dhcp6.domain-search"; "dhcp6.name-servers"]
  else if emit && not ipv6 then
    debug "%s is the DNS interface" interface ;
    ["domain-name"; "domain-name-servers"]
  else
    debug "%s is NOT the DNS interface" interface ;
    []
```